### ISSUE TITLE
`derive_filter? false` does not appear to do anything at the resource level

### ISSUE THREAD

**Issue by sevenseacat** [ash-project/ash_json_api#193]
The [docs](https://hexdocs.pm/ash_json_api/dsl-ashjsonapi-resource.html#json_api-derive_filter?) say this should be supported at the resource level, but after adding it to my `json_api` block:

```elixir
  json_api do
    type "artist"
    includes [:albums]

    derive_filter? false
  end
```

I still have `filter` appearing in Swagger UI:

<img width="684" alt="Screenshot 2024-07-02 at 8 59 01 PM" src="https://github.com/ash-project/ash_json_api/assets/543859/ef88a5e1-47b8-480e-b89c-f445aa487914">

Adding it to an individual route (ie. `index :search`) *does* work as expected.

In `open_api.ex` there is one reference to `route.derive_filter?` that looks a bit sketchy to me:

```elixir
    defp has_index_route?(domains, resource) do
      resource
      |> AshJsonApi.Resource.Info.routes(domains)
      |> Enum.any?(fn route ->
        route.type == :index && read_action?(resource, route) && route.derive_filter?
      end)
    end
```

Does this need to be `route.derive_filter? || AshJsonApi.Resource.Info.derive_filter?(resource)` ?

**Comment by zachdaniel** [comment]
Yep, it does :)

**Comment by zachdaniel** [comment]
Actually now I'm not sure...that probably does need to change, but it wouldn't solve the issue I don't think.

**Comment by zachdaniel** [comment]
We should have tests, yes. And there are tests, but it's not 100% coverage, as you're discovering. but for example, in `/open_api_test.exs`

```elixir
      %OpenApiSpex.Operation{} = operation = api_spec.paths["/authors/no_filter"].get
      refute Enum.any?(operation.parameters, &(&1.name == :filter))
```

**Comment by zachdaniel** [comment]
Hm... as far as I can tell, we are always checking in both places...

okay, so `has_index_route?` didn't need to be `&& derive_filter?(resource)` because it was already inside a function that was checking that.

**Comment by zachdaniel** [comment]
Got it reproduced in a test. We were only testing the route specific stuff.

**Comment by zachdaniel** [comment]
alas, twas a typo.

**Comment by zachdaniel** [comment]
Typo fix here: b1aae808a3038f7e7e1facf4b366c979c03e606a

**Comment by sevenseacat** [comment]
This works as expected now, thank you :D
